timemachine/fe/single_topology.py

```diff
@@ -1158,81 +906,105 @@ def assert_torsions_defined_over_non_linear_angles(system: GuestSystem | HostGue
                 )
 
 
-def assert_chiral_consistency(src_chiral_idxs: NDArray, dst_chiral_idxs: NDArray):
-    """
-    Assert that the chiral volumes are not inverting in src and dst.
-    """
-    src_chiral_restr_idx_set = ChiralRestrIdxSet(src_chiral_idxs)
-    dst_chiral_restr_idx_set = ChiralRestrIdxSet(dst_chiral_idxs)
-
-    # ensure that we don't have any chiral inversions between src and dst end states
-    assert len(src_chiral_restr_idx_set.allowed_set.intersection(dst_chiral_restr_idx_set.disallowed_set)) == 0
-    assert len(dst_chiral_restr_idx_set.allowed_set.intersection(src_chiral_restr_idx_set.disallowed_set)) == 0
-
+def build_combined_atom_map_mixin(
+    num_host_atoms: int, mol_a: Chem.Mol, mol_b: Chem.Mol, core: NDArray[np.int32]
+) -> AtomMapMixin:
+    new_core_ = []
+    for idx in range(num_host_atoms):
+        new_core_.append((idx, idx))
 
-@dataclass
-class AlignedPotential:
-    idxs: NDArray[np.int32]
-    src_params: NDArray[np.float64]
-    dst_params: NDArray[np.float64]
-    mins: NDArray[np.float64]
-    maxes: NDArray[np.float64]
+    for i, j in core:
+        new_core_.append((i + num_host_atoms, j + num_host_atoms))
 
-    def interpolate(self, lamb):
-        raise NotImplementedError()
+    new_core = np.array(new_core_, dtype=np.int32)
 
+    num_atoms_a = num_host_atoms + mol_a.GetNumAtoms()
+    num_atoms_b = num_host_atoms + mol_b.GetNumAtoms()
 
-class AlignedBond(AlignedPotential):
-    def interpolate(self, lamb):
-        k_min = 0.1
-        params = batch_interpolate_harmonic_bond_params(
-            self.src_params, self.dst_params, lamb, k_min, self.mins, self.maxes
-        )
-        params = jnp.array(params).T
+    return AtomMapMixin(num_atoms_a, num_atoms_b, new_core)
 
-        return HarmonicBond(self.idxs).bind(params)
 
+def build_end_state_reference_all_pairs(
+    num_guest_atoms: int, host_nonbonded_all_pairs: BoundPotential[Nonbonded]
+) -> BoundPotential[Nonbonded]:
+    host_params = host_nonbonded_all_pairs.params
+    nb_params = np.concatenate([host_params, np.zeros((num_guest_atoms, host_params.shape[1]))])
```

Comment:
(keep the comment about why we have 0 params here): 

\# Note: The choice of zeros here is arbitrary. It doesn't affect the
\# potentials or grads, but any function like the seed could depend on these values.

Reply:
comment added in https://github.com/proteneer/timemachine/pull/1538/commits/097ef98248ad6fe6998af8d81514f91519709b36